packages/roosterjs-content-model-plugins/lib/autoFormat/AutoFormatPlugin.ts

```diff
@@ -100,6 +102,36 @@ export class AutoFormatPlugin implements EditorPlugin {
         this.editor = null;
     }
 
+    private shouldHandleInputEventExclusively(editor: IEditor, event: EditorInputEvent) {
+        const rawEvent = event.rawEvent;
+        const selection = editor.getDOMSelection();
+        if (
+            rawEvent.inputType === 'insertText' &&
+            selection &&
+            selection.type === 'range' &&
+            selection.range.collapsed &&
+            rawEvent.data == ' '
+        ) {
+            const previousText = selection.range.startContainer.textContent?.trim();
+            if (!previousText) {
+                return false;
+            }
+
+            return isLastWordUrl(previousText) || isLetterFollowedByMarker(previousText);
```

Comment:
Ideally, we should have a same check when checking exclusiveness and really handle the event, rather than having separate functions. 

But if that is too hard to refactor, or have bad perf impact, then it is ok to keep them separate. 

Can you double check?

Reply:
I changed the code to use the getListTypeStyle and promoteLink, if these functions do have return, we should handle the event exclusively